Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/controller/settings/ExperimentalSettingsController.java

```diff
@@ -77,9 +132,133 @@ private void setupConcurrentFileDownloadingChunksSetting(SettingsGroup group) {
         ) {
             @Override
             public String getBottomDescription() {
-                return getString(R.string.settings_concurrent_file_downloading_description) + "\n\n" + items.get(
-                        selected).name;
+                return getString(R.string.settings_concurrent_file_downloading_description)
+                        + "\n\n" + items.get(selected).name;
             }
         }));
     }
+
+    private void setupZonesResetButton(SettingsGroup group) {
+        resetExclusionZonesSetting = new LinkSettingView(
+                this,
+                R.string.setting_exclusion_zones_reset_zones,
+                R.string.setting_exclusion_zones_reset_zones_description,
+                (v) -> {
+                    if (!isAndroid10()) {
+                        showToast(context, R.string.setting_exclusion_zones_can_only_be_used_on_android_10);
+                        return;
+                    }
+
+                    exclusionZonesHolder.resetZones();
+                    ((StartActivity) context).restartApp();
+                }
+        );
+
+        requiresUiRefresh.add(group.add(resetExclusionZonesSetting));
+    }
+
+    private void setupZonesEditor(SettingsGroup group) {
+        gestureExclusionZonesSetting = new LinkSettingView(
+                this,
+                R.string.setting_exclusion_zones_editor,
+                R.string.setting_exclusion_zones_editor_description,
+                (v) -> showZonesDialog()
+        );
+
+        requiresUiRefresh.add(group.add(gestureExclusionZonesSetting));
+    }
+
+    private void showZonesDialog() {
+        if (!isAndroid10()) {
```

Comment:
It's there because onOptionClicked requires Android Q to be called.

Reply:
What I mean is that showZonesDialog is only called from the setting view, and that setting view is disabled if you're not on Android 10.

Reply:
Yeah I know, but Android Studio doesn't like it. To fix that I will have to remove `@RequiresApi(Build.VERSION_CODES.Q)` from `AdjustAndroid10GestureZonesController` and I really don't want to do that because it helps preventing the controller getting called occasionally on a non Android 10 device.